src/components/AccountModule/WalletConnectionModule.js

```diff
@@ -0,0 +1,240 @@
+import React, { useEffect, useRef, useState } from 'react'
+import PropTypes from 'prop-types'
+import {
+  ButtonBase,
+  EthIdenticon,
+  GU,
+  IconDown,
+  Popover,
+  RADIUS,
+  springs,
+  textStyle,
+  useTheme,
+  useViewport,
+  unselectable,
+} from '@aragon/ui'
+import { animated, Spring } from 'react-spring'
+import { shortenAddress } from '../../web3-utils'
+import { useLocalIdentity } from '../../hooks'
+import { useSyncInfo } from './useSyncInfo'
+import { useWallet } from '../../wallet'
+import WalletNotConnected from './WalletNotConnected'
+import WalletConnectionInfo from './WalletConnectionInfo'
+import { useNetworkConnectionData, useWalletConnectionDetails } from './utils'
+
+// Metamask seems to take about ~200ms to send the connected accounts.
+// This is to avoid a flash with the connection button.
+const ACCOUNT_MODULE_DISPLAY_DELAY = 500
+
+const AnimatedDiv = animated.div
+
+function WalletConnectionModule({ locator, compact }) {
+  const { isConnected } = useWallet()
+  const [display, setDisplay] = useState(false)
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setDisplay(true)
+    }, ACCOUNT_MODULE_DISPLAY_DELAY)
+
+    return () => clearTimeout(timer)
+  }, [])
+
+  if (!display) {
+    return null
+  }
+
+  return (
+    <Spring
+      from={{ opacity: 0, scale: 0.96 }}
+      to={{ opacity: 1, scale: 1 }}
+      config={springs.swift}
+      native
+    >
+      {({ opacity, scale }) => (
+        <AnimatedDiv
+          style={{
+            opacity,
+            transform: scale.interpolate(v => `scale3d(${v}, ${v}, 1)`),
+          }}
+          css={`
+            display: flex;
+            height: 100%;
+            align-items: center;
+          `}
+        >
+          {isConnected ? (
+            <WalletConnectedMode locator={locator} />
+          ) : (
+            <WalletNotConnected compact={compact} />
```

Comment:
So I'm wondering... given that we basically have two parts, with the second part also having two modes based on the connected state:

1. Connect account
2. Connection information / account information (and their related popovers)

It may make sense to avoid duplicating all this animation and popover layout and consolidate it into one.

Rather than having this conditional in `OrgView`:

```js
<WalletConnectionModule locator={locator} />
{!isConnected && <ClientConnectionModule locator={locator} />}
```

We simply keep rendering a single `AccountModule`, and in `AccountModule`, render (changing `WalletNotConnected` to `EnableWallet`):

```js
{!isConnected && <EnableWallet />}
<div ref={containerRef}>
  {isConnected ? <WalletConnectedButton /> : <ConnectionStatusButton />}
  <Popover>
    {isConnected ? <WalletConnectedPopover /> : <ConnectionStatusPopover />}
  </Popover>
</div>
```

I think `WalletConnectedButton` and `ConnectionStatusButton` could even be combined, because they change modes based on just the state from `useWallet()`.

`WalletConnectedPopover` and `ConnectionStatusPopover` may be harder and we can see if it makes sense to consolidate these.